js/src/redux/reducers/snippetEditor.js

```diff
@@ -41,15 +43,34 @@ function snippetEditorReducer( state = INITIAL_STATE, action ) {
 			};
 
 		case UPDATE_REPLACEMENT_VARIABLE:
-			return {
-				...state,
-				replacementVariables: [
-					...state.replacementVariables,
-					{
+			const index = state.replacementVariables.findIndex( replaceVar => {
```

Comment:
This `const` and the `let` below within a `case` clause, don't make ESLint happy, which errors with:
`Unexpected lexical declaration in case block           no-case-declarations`
More details: https://eslint.org/docs/rules/no-case-declarations